core/templates/domain/exploration/AnswerGroupObjectFactory.ts

```diff
@@ -79,7 +79,7 @@ export class AnswerGroupObjectFactory {
   createNew(
       rules: Rule[], outcome: Outcome,
       trainingData: readonly InteractionAnswer[],
-      taggedSkillMisconceptionId: string): AnswerGroup {
+      taggedSkillMisconceptionId: string | null): AnswerGroup {
```

Comment:
Why can this be `null`? Is it possible to easily refactor this to not use `null`?

Reply:
@vojtechjelinek Take a look at this https://github.com/oppia/oppia/pull/12561#discussion_r629091485. Also in files like `QuestionObjectFacotry` & `state-respone.directive.ts`  we check this property to be not null.So I think it won't be easy to make it not use `null` and  this should be the correct type.

Reply:
It is correct that this may be null. But, in all cases where null is used as a possible type, please leave a line comment explaining why this is possible and what it means. Probably the best place to do it is in the relevant interface, similar to what you did for StoryContents.